streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -631,6 +625,14 @@ public void setStreamsUncaughtExceptionHandler(final java.util.function.Consumer
         this.streamsUncaughtExceptionHandler = streamsUncaughtExceptionHandler;
     }
 
+    public void maybeSendShutdown() {
+        if (assignmentErrorCode.get() == AssignorError.SHUTDOWN_REQUESTED.code()) {
+            log.warn("Detected that shutdown was requested. " +
+                    "All clients in this app will now begin to shutdown");
+            mainConsumer.enforceRebalance();
```

Comment:
Since this thread is going to immediately shut down anyways, I think we can skip the `mainConsumer.enforceRebalance()`

Reply:
This is also how we trigger the rebalance for the other threads so we can't just remove it

Reply:
But do we even need to invoke `maybeSendShutdown` at all from the two catch blocks? The thread that we start up should handle this

Reply:
No we don't we can take those out